lib/index.js

```diff
@@ -262,9 +265,47 @@ module.exports = class DockerComposeParser {
   }
 
   /**
+   * Adds a main service if one doesn't already exist to the mains.builds object, and adds it to the services array
+   *
+   * @param {Service[]} services                  - Array of services
+   * @param {Object}    mains                     - Object containing all of the main builds and externals
+   * @param {Object}    mains.externals           - All services with external (github) builds (indexed by name)
+   * @param {Object}    mains.builds              - All services with local builds (same repo as compose file) (indexed by name)
+   * @param {Object}    opts
+   * @param {String}    opts.repositoryName       - Name of repository in GH or other service
+   * @param {String}    opts.userContentDomain    - User content domain for API environment. Used for creating hostnames.
+   * @param {String}    opts.ownerUsername        - Organization name in Runnable. Used for creating hostnames.
+   * @param {String}    opts.scmDomain            - Domain for used SCM. E.g. github.com
+   * @private
+   */
+  static addMainIfMissing (services, mains, opts) {
```

Comment:
Yes, 2 actually.  So I removed those (since the logic is pretty much all different) and made this one from them